bridge-stp.in

```diff
@@ -52,7 +58,7 @@ case $2 in
         if [ -e "$conf_file" ]; then
             . "$conf_file" || exit
         fi
-        checkpid $pid_file || exit 1
+        checkpid || exit 1
```

Comment:
`/var/run` is cleaned up on reboot on many systems.  However, it is not guaranteed to be cleaned up on reboot.  For example, I have a number of RHEL5 and RHEL6 systems that don't clean it up on reboot (It is possible my systems are misconfigured somehow, but I /var/run is not mounted as tmpfs, I don't see any startup scripts to clean it up, and I've confirmed that it is not cleaned up on reboot).

As you mentioned, segfaults are another case we need to handle gracefully.  I don't think there is any "proper" way to handle PID files after a crash.

So, unless there is a better way to verify that an arbitrary process is really an mstpd process, I think this additional PID checking is necessary...